lib.nix

```diff
@@ -0,0 +1,37 @@
+{ lib }:
+let
+  inherit (import ./semver.nix { inherit lib; }) satisfiesSemver;
+
+  # Returns true if pythonVersion matches with the expression in pythonVersions
+  isCompatible = pythonVersion: pythonVersions: let
+    operators = {
+      "||" = cond1: cond2: cond1 || cond2;
+      "," = cond1: cond2: cond1 && cond2; # , means &&
+    };
+    # split string at "," and "||"
+    tokens = builtins.filter (x: x != "") (builtins.split "(,|\\|\\|)" pythonVersions);
+  in
+    (
+      builtins.foldl' (
+        acc: v: let
```

Comment:
try moving the `acc: v:` function into the parent `let` binding, it's going to make it more readable